### PR TITLE
Clarify and fix name/address validation constraints in documentation and error messages

### LINKED ISSUE #192: [PE-D][Tester B] UG: Missing Parameter Constraints for Address Field in User Guide
## Bug Description
The User Guide fails to document the validation constraints enforced by the application for the `Address` parameter in the `add` and `edit` commands.

### Actual Behavior
In the "Features" section for the `add` command, the parameter is listed as `a/ADDRESS`. There is no mention of the required constraints that make a valid address.

### Expected Behavior 
The User Guide should explicitly list the constraints for the Address field so users know exactly what is acceptable. For example: `Address: Can take any values, but it cannot be blank and the first character must not be a whitespace.`

## Impact
While the application will eventually return an error message explaining the constraint if a command fails, users should not have to rely on a trial-and-error approach to learn basic formatting rules. The primary purpose of a User Guide is to equip the user with the necessary information before they execute a command. Forcing users to discover these constraints reactively through application errors hinders the users and slows down their workflow. Explicitly documenting that the address cannot start with a space prevents interruptions in workflows and improves the overall user experience.

-------------
Labels: `severity.Low` `type.DocumentationBug`

### LINKED ISSUE #245: [PE-D][Tester A] Names beginning with special characters incorrectly rejected despite being documented as valid
The User Guide states that names may contain hyphens, apostrophes, and asterisks, yet the application rejects any name that begins with one of these characters with the error message "Names should only contain alphanumeric characters and spaces, and it should not be blank.", which is arguably an incorrect message to display in this case as well.

Image

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by verudex**
Fixes #245 
Fixes #192 

## Description
This pull request addresses critical mismatches between the validation logic and documentation for name and address fields. The changes ensure that users can understand input constraints from error messages and documentation without requiring trial and error.

## Changes
**Name Validation (model, docs, tests):**
- Updated the name validation error message to explicitly state that separators (space, hyphen, apostrophe, comma) are only allowed between alphanumeric parts and cannot be consecutive
- Clarified in the User Guide that names must start and end with alphanumeric characters and cannot have consecutive separators
- Added comprehensive test cases covering leading separator, trailing separator, and consecutive separator edge cases
- Error message now fully explains the constraint: "Names should be 1 to 100 characters long, start and end with an alphanumeric character, and use only alphanumeric characters separated by a single space, hyphen (-), apostrophe ('), or comma (optionally followed by one space)."

**Address Validation (model, docs):**
- Updated the address validation error message to explicitly mention that addresses must be 1 to 200 characters, cannot be blank, require single spaces between words, and list all allowed special characters including backslash
- Documented address constraints in the User Guide with clear examples of valid and invalid inputs

**User Guide Updates:**
- Split combined allergy/condition constraints into separate sections for `al/` and `mc/` with field-specific limits and examples
- Added explicit name format constraint section explaining separator rules and providing invalid examples
- All field constraints now follow a consistent format: allowed character set, length limits, and separate valid/invalid examples

## Motivation
Users were unable to determine why valid-looking names or addresses were rejected because:
1. Error messages were vague and did not document the separator placement or consecutiveness rules
2. User Guide documentation did not clarify that separators cannot be used at the start/end or consecutively
3. Allergy and condition constraints were combined, making field-specific limits unclear

These changes provide clear, actionable constraint documentation so users can format input correctly on their first attempt.

## Testing
- Extended NameTest with new edge cases for leading/trailing/consecutive separators
- All existing tests pass with updated error message text
- Updated documentation examples match actual validation behavior

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -141,14 +141,19 @@ Format: `add n/NAME ic/NRIC x/SEX dob/DOB p/PHONE_NUMBER e/EMAIL a/ADDRESS [al/A
 
 **:information_source: Accepted name formats**<br>
 
-Names are limited to 100 characters. DoctorWho currently accepts the following special characters in the patient's name:
+For both `add` and `edit`, names are limited to **1 to 100 characters**.
+Names must start and end with an alphanumeric character.
+DoctorWho accepts the following separators only **between** alphanumeric name parts:
+only one separator is allowed at a time (i.e., no consecutive separators such as `- `, `''`, or double spaces).
```
Comment (alieron) [inline docs/UserGuide.md]: What about `, ` though? we allow it in `Henry, Tan`. Also the regex enforces start must be alnum, but not the end.

File: src/test/java/doctorwho/model/patient/NameTest.java [hunk 2]
```diff
@@ -31,6 +31,10 @@ public void isValidName() {
         assertFalse(Name.isValidName("peter*")); // contains invalid characters
         assertFalse(Name.isValidName("Jäger")); // contains diacritics
         assertFalse(Name.isValidName("Ali s/o Ahmad")); // contains slash
+        assertFalse(Name.isValidName("-David")); // starts with separator
+        assertFalse(Name.isValidName("David-")); // ends with separator
```
Comment (alieron) [inline src/test/java/doctorwho/model/patient/NameTest.java:35]: Really? 👀

**Review by alieron — CHANGES_REQUESTED** [review]
Minor changes

File: docs/UserGuide.md (hunk 1 above)
Comment (verudex) [reply docs/UserGuide.md]: Fixed the clarification for the comma. Apparently the end is enforced as well. Reference below:

Add command:
image

Edit command:
image

**Review by alieron — APPROVED** [review]
LGTM
